typing/env.ml

```diff
@@ -824,14 +827,41 @@ let find proj1 proj2 path env =
 
 let find_value =
   find (fun env -> env.values) (fun sc -> sc.comp_values)
-and find_type_full =
-  find (fun env -> env.types) (fun sc -> sc.comp_types)
 and find_modtype =
   find (fun env -> env.modtypes) (fun sc -> sc.comp_modtypes)
 and find_class =
   find (fun env -> env.classes) (fun sc -> sc.comp_classes)
 and find_cltype =
   find (fun env -> env.cltypes) (fun sc -> sc.comp_cltypes)
+and proto_find_type =
+  find (fun env -> env.types) (fun sc -> sc.comp_types)
+
+let normalize_path_ext p env =
+  let module S = Set.Make(Path) in
+  let rec normalize s p env =
+  match fst @@ proto_find_type p env with
+    | exception Not_found -> p
+    | { type_kind=Type_open|Type_abstract;
+        type_manifest = Some { desc= Tconstr(p,_,_) ; _ } ; _ }
+      when not (S.mem p s) -> normalize (S.add p s) p env
+    | _ -> p in
+  normalize S.empty p env
+
+let find_type_full p env =
+    let tyd, (constrs, lbls) as classical = proto_find_type p env in
+    (* Extensible type constructors are not caught by proto_find_type,
+       and require the following specialized code path. *)
+    match tyd.type_kind with
+    | Type_variant _ | Type_record _ -> classical
+    | Type_open | Type_abstract ->
```

Comment:
The `Type_abstract` case is not necessary here anymore since #1253. (It was when this PR was written.)